bindings/guile/test/test-engine-extras.scm

```diff
@@ -239,42 +241,46 @@
           (pricedb? #t)    ; boolean: add pricedb entry?
           void-reason      ; string: void-reason (def = not-voided)
           reconcile        ; pair  : (cons reconciled reconciled-date)
+          currency         ; currency
           num              ; string: num field   (def = null)
           notes            ; string: notes       (def = null)
           memo)            ; string: memo        (def = null)
   (and (pair? list-of-splits)
        (let* ((book (gnc-get-current-book))
               (txn (xaccMallocTransaction book))
-              (first-split (vector-ref (car list-of-splits) 0)))
+              (first-split (vector-ref (car list-of-splits) 0))
+              (txn-curr (or currency (xaccAccountGetCommodity first-split))))
          (xaccTransBeginEdit txn)
          (xaccTransSetDescription txn (or description (env-string env "ponies")))
-         (xaccTransSetCurrency txn (xaccAccountGetCommodity first-split))
+         (xaccTransSetCurrency txn txn-curr)
          (xaccTransSetDate txn DD MM YY)
          (for-each
           (lambda (split)
             (let ((acc (vector-ref split 0))
                   (val (vector-ref split 1))
                   (amt (vector-ref split 2))
+                  (action (and (> (vector-length split) 3)
+                               (vector-ref split 3)))
                   (newsplit (xaccMallocSplit book)))
               (xaccSplitSetParent newsplit txn)
               (xaccSplitSetAccount newsplit acc)
               (xaccSplitSetValue newsplit val)
               (xaccSplitSetAmount newsplit amt)
-              (if num (gnc-set-num-action txn newsplit num num))
-              (if memo (xaccSplitSetMemo newsplit memo))
+              (if memo   (xaccSplitSetMemo newsplit memo))
+              (if action (xaccSplitSetAction newsplit action))
```

Comment:
That's not erroneous. There's a book option to use the split Action field for numbers so that you can have different numbers in different accounts on the same transaction. `gnc-set-num-action` reads the preference and puts the number in the appropriate field.

Reply:
I know and not sure how to engineer this test utility function. I find the split-action book property and API extremely inelegant and wish to enable it completely as a book upgrade*. The acb report uses the split action property to denote fees.

* TBD in devel I guess.

Reply:
I suppose you mean "remove it" rather than "enable it", it's already enabled. Expect wailing and gnashing of teeth, there are users who think it's a Good Thing.

As for using it to designate a fee split I suppose it matters in what account. If it's Expenses:Investment:Commisisons and Fees then maybe it's unlikely that a user would want to put a number there.

Reply:
I meant "enable and enforce it". ISTM if enabled then num becomes t-num and action becomes num, therefore more fields become available. Therefore they'll be no data loss. But whether to display num vs action in the register column would depend on the preference.

Reply:
Sorry, you've lost me completely, but I haven't looked hard at the code in years. I thought that it hid the transaction object's num data in basic mode and pointed the  register column at the current account's split's action field; split view shows everything in its usual place. Do I mis-remember?